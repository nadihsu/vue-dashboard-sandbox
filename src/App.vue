<template lang="pug">
.wrapper
  el-container.main-layout
    el-aside(:width="lang === 'en-US' ? '11rem' : '9rem'")
      sidebar.
    el-container
      el-header
        el-menu(
        :default-active="activeIndex"
        class="el-menu-demo"
        mode="horizontal"
        :ellipsis="false")
          el-menu-item LEGO
          .flex-grow
          .right-menu-content
            el-dropdown(@command="handleLangChange")
              span.el-dropdown-link {{ $t('M_LANG') }}
                el-icon.el-icon--right
                  i-ep-arrow-down
              template(#dropdown)
                el-dropdown-item(command="zh-TW") 繁體中文
                el-dropdown-item(command="en-US") EN
      el-main.main-content
        router-view.
</template>

<script>
import { storeToRefs } from 'pinia';
import { useI18n } from 'vue-i18n';
import useLangStore from '@/store/useLangStore';
import Sidebar from './components/Sidebar.vue';

export default {
  components: { Sidebar },
  setup() {
    const { t } = useI18n();
    const { lang } = storeToRefs(useLangStore());
    const { updateLang } = useLangStore();

    /**
     * 切換語系
     *
     * @param {string} value 語系代碼
     */
    function handleLangChange(value) {
      updateLang(value);
    }

    return {
      t,
      lang,
      handleLangChange,
    };
  },
};
</script>

<style lang="scss">
@import 'https://cdnjs.cloudflare.com/ajax/libs/normalize/8.0.1/normalize.min.css';
@import './styles/main.scss';

.wrapper {
  height: 100vh;
  display: flex;
  flex-direction: row;
}

.right-menu-content {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  flex-grow: 1;
}

.main-layout {
  display: flex;
  height: 100vh;
  flex: 1;

  aside {
    position: relative;
    height: 100%;
    padding: 1rem;
    background-color: #F6F6F6;
  }

  .main-content {
    overflow-y: scroll;
  }
}

.flex-grow {
  flex-grow: 1;
}

.el-dropdown-link:focus-visible {
  outline: none;
}
</style>
