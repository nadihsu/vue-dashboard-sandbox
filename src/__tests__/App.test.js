import { render } from 'Tests/unit/TestUtils';
import App from '@/App.vue';

describe('Main Page', () => {
  it('基本', () => {
    const setup = render(App);

    expect(setup.html()).toMatchSnapshot();
  });
});
