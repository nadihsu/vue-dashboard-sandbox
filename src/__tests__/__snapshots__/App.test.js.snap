// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Main Page > 基本 1`] = `
<div class="wrapper">
  <section class="el-container main-layout">
    <aside class="el-aside" style="--el-aside-width: 9rem;">
      <a class href="#/">SIDEBAR.USER</a>
      <a class href="#/about">SIDEBAR.ABOUT</a>
    </aside>
    <section class="el-container is-vertical">
      <header class="el-header">
        <ul class="el-menu el-menu--horizontal el-menu-demo" role="menubar" style="--el-menu-level: 0;">
          <li class="el-menu-item" role="menuitem" tabindex="0">LEGO</li>
          <div class="flex-grow" tabindex="0"></div>
          <div class="right-menu-content" tabindex="0">
            <div class="el-dropdown">
              <span aria-controls aria-expanded="false" aria-haspopup="menu" class="el-dropdown-link el-tooltip__trigger el-tooltip__trigger" id role="button" tabindex="0">M_LANG<i class="el-icon el-icon--right">
                  <svg height="1.2em" viewBox="0 0 1024 1024" width="1.2em">
                    <path d="M831.872 340.864L512 652.672L192.128 340.864a30.592 30.592 0 0 0-42.752 0a29.12 29.12 0 0 0 0 41.6L489.664 714.24a32 32 0 0 0 44.672 0l340.288-331.712a29.12 29.12 0 0 0 0-41.728a30.592 30.592 0 0 0-42.752 0z" fill="currentColor" />
                  </svg>
                </i>
              </span>
              <!--teleport start-->
              <!--teleport end-->
              <!--v-if-->
            </div>
          </div>
        </ul>
      </header>
      <main class="el-main main-content">
        <!---->
      </main>
    </section>
  </section>
</div>
`;
